cherry_picker/cherry_picker/cherry_picker.py

```diff
@@ -422,10 +417,37 @@ def get_base_branch(cherry_pick_branch):
     """
     return '2.7' from 'backport-sha-2.7'
     """
-    prefix, sep, base_branch = cherry_pick_branch.rpartition('-')
+    prefix, sha, base_branch = cherry_pick_branch.split('-', 2)
+
+    if prefix != 'backport':
+        raise ValueError('branch name is not prefixed with "backport-".  Is this a cherry_picker branch?')
+
+    if not re.match('[0-9a-f]{7,40}', sha):
+        raise ValueError(f'branch name has an invalid sha: {sha}')
+
+    cmd = ['git', 'log', '-r', sha]
```

Comment:
I thought about this but something struck me as off about doing that and I couldn't quite put my finger on it yesterday.  This morning I realized it's because check_repo() is not well encapsulated.  The sha is contained as an attribute on the CherryPicker object but the repository is coming from the current working directory.  This wasn't immediately obvious because cherry_picker doesn't give you the option to work on a repository that's not the current working directory but when I thought about making the code generic to both places it triggered my sense that something was wrong.

Now that I know that's what struck me as wrong, I'd like to go ahead and move this to its own function in this PR and submit a second PR that allows setting the working directory from which all the git commands are run.  When I do that, though, functions like get_base_branch() will either need to become methods of CherryPicker or I'll have to pass the path to the repository all the way through the stack.  Does that future plan sound okay?

Reply:
Sounds good